mayastor/src/bdev/nexus/nexus_bdev.rs

```diff
@@ -752,299 +728,9 @@ impl Nexus {
-        let results = channels
-            .writers
-            .iter()
-            .map(|c| unsafe {
-                let (b, c) = c.io_tuple();
-                spdk_bdev_write_zeroes_blocks(
-                    b,
-                    c,
-                    io.offset() + io.nexus_as_ref().data_ent_offset,
-                    io.num_blocks(),
-                    Some(Self::io_completion),
-                    io.as_ptr() as *mut _,
-                )
-            })
-            .collect::<Vec<_>>();
-
-        self.check_io_submission(&results, &io);
-    }
-
-    pub(crate) fn nvme_admin(&self, io: &Bio, channels: &NexusChannelInner) {
-        if io.nvme_cmd().opc() == nvme_admin_opc::CREATE_SNAPSHOT as u16 {
-            // FIXME: pause IO before dispatching
-            debug!("Passing thru create snapshot as NVMe Admin command");
-        }
-        // for replicas, passthru only works with our vendor commands as the
-        // underlying bdev is not nvmf
-        let results = channels
-            .writers
-            .iter()
-            .map(|c| unsafe {
-                debug!("nvme_admin on {}", c.get_bdev().driver());
-                if c.get_bdev().driver() == "lvol" {
-                    // Local replica, vbdev_lvol does not support NVMe Admin
-                    // so call function directly
-                    let lvol = Lvol::try_from(c.get_bdev()).unwrap();
-                    match io.nvme_cmd().opc() as u8 {
-                        nvme_admin_opc::CREATE_SNAPSHOT => {
-                            subsys::create_snapshot(
-                                lvol,
-                                &io.nvme_cmd(),
-                                io.as_ptr(),
-                            );
-                        }
-                        _ => {
-                            error!(
-                                "{}: Unsupported NVMe Admin command {:x}h from IO {:?}",
-                                io.nexus_as_ref().name,
-                                io.nvme_cmd().opc(),
-                                io.as_ptr()
-                            );
-                            Self::io_completion_local(
-                                false,
-                                io.as_ptr().cast(),
-                            );
-                        }
-                    }
-                    return 0;
-                }
-                let (desc, chan) = c.io_tuple();
-                spdk_bdev_nvme_admin_passthru(
-                    desc,
-                    chan,
-                    &io.nvme_cmd(),
-                    io.nvme_buf(),
-                    io.nvme_nbytes(),
-                    Some(Self::io_completion),
-                    io.as_ptr() as *mut _,
-                )
-            })
-            .collect::<Vec<_>>();
-
-        self.check_io_submission(&results, &io);
+    pub fn io_completion_local(_success: bool, _parent_io: *mut c_void) {
```

Comment:
Is this function used anywhere ?

Reply:
It called in the lvol layer and I did not want to remove all the code when we enable it again. So i left it this way for now.